pkg/provider/github/github.go

```diff
@@ -177,3 +186,76 @@ func (v *Provider) getObject(ctx context.Context, sha string, runevent *info.Eve
 	}
 	return decoded, err
 }
+
+// Detect processes event and detect if it is a github event, whether to process or reject it
+// returns (if is a GH event, whether to process or reject, error if any occurred)
+func (v *Provider) Detect(reqHeader *http.Header, payload string, logger *zap.SugaredLogger) (bool, bool, error) {
```

Comment:
this is returns
isGH - which will tell whether to check for other provider or not
processReq - whether to process or reject
wdyt?

Reply:
yeah it's fine !